src/components/Input.react.js

```diff
@@ -24,294 +24,297 @@ function formatValue(val, type){
-            if (updatemode === 'blur') {
-              this.setState({value: newValue});
-              if (setProps) {
-                  setProps({value: newValue});
-              }
-              if (fireEvent) {fireEvent({event: 'blur'});}
-            }
-          }
-        }
-        onChange={
-          (e) => {
-            const newValue = formatValue(e.target.value, type)
-            this.setState({value: newValue});
-    				if (updatemode === 'keypress') {
-              if (setProps) {
-                  setProps({value: newValue});
-              }
-    				}
-    				if (fireEvent) {fireEvent({event: 'change'});}
-  				}
-        }
-        onKeyPress={
-          (e) => {
-            if (type === 'number'){
-              const curVal = this.state.value;
-              const keyCode = e.keyCode || e.which;
-              const keyValue = String.fromCharCode(keyCode)
-              const unicodePlus = 43;
-              const unicodeNine = 57;
-            if ((keyCode <= unicodePlus || keyCode > unicodeNine) || (keyValue === '/'))
-              {e.preventDefault();}
-            if (( keyValue === '-') && (curVal.length > 0))
-              {e.preventDefault();}
-            if ((keyValue === '.' || keyValue === ',') && (/[.]|,/.test(curVal.value)))
-              {e.preventDefault();}
-            }
-          }
-        }
-        onClick = {
-          (e) => {
-            const newValue = formatValue(e.target.value, type)
-            this.setState({value: newValue});
-            if (setProps) {
-              setProps({value: newValue});
-            }
-          }
-        }
-        value={value}
-        {...omit(['fireEvent', 'setProps', 'value', 'updatemode'], this.props)}
-        />
-    );
-	}
+export default class Input extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            value: formatValue(props.value, props.type),
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState({value: formatValue(nextProps.value, nextProps.type)});
+    }
+
+    render() {
+        const {fireEvent, setProps, type, updatemode} = this.props;
+
+        const {value} = this.state;
+        return (
+            <input
+                onBlur={() => {
+                    const newValue = formatValue(this.state.value, type);
+                    if (updatemode === 'blur') {
+                        this.setState({value: newValue});
+                        if (setProps) {
+                            setProps({value: newValue});
+                        }
+                        if (fireEvent) {
+                            fireEvent({event: 'blur'});
+                        }
+                    }
+                }}
+                onChange={e => {
+                    const newValue = formatValue(e.target.value, type);
+                    this.setState({value: newValue});
+                    if (updatemode === 'keypress') {
```

Comment:
Similarly here - this `if` only works if `updatemode` is a string, not when it's an array.